receiver/envoyalsreceiver/internal/als/server.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package als // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/envoyalsreceiver/internal/als"
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	alsv3 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v3"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+)
+
+type Server struct {
+	nextConsumer consumer.Logs
+	obsrep       *receiverhelper.ObsReport
+}
+
+func New(nextConsumer consumer.Logs, obsrep *receiverhelper.ObsReport) *Server {
+	return &Server{
+		nextConsumer: nextConsumer,
+		obsrep:       obsrep,
+	}
+}
+
+func (s *Server) StreamAccessLogs(logStream alsv3.AccessLogService_StreamAccessLogsServer) error {
+	for {
+		data, err := logStream.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		ctx := s.obsrep.StartLogsOp(context.TODO())
```

Comment:
```suggestion
		ctx := s.obsrep.StartLogsOp(context.Background())
```
Unless we have a clear path to actually implementing this, we should just use `context.Background`. I can't think of any obvious way we'll be able to get a context here given this method is required to implement the `AccessLogServiceServer` interface.